### DISCUSSION TITLE
A high number of lawyer registration requests are not reaching NyayMitra

### DISCUSSION THREAD

**Discussion by Ramu-kandimalla** [egovernments/Digit-Core#613] [Bugs]
A high number of lawyer registration requests are not reaching NyayMitra. They are unable to re-register as their Bar Id is already stored.

**Analysis:** 

Hi @atulgupta2024 @subhashini-egov , this is happening because sometimes indexer not able to enrich advocate details with latest process instance history , because it searched through workflow service (database search) and the workflow data is not persisted yet cc: @Ramu-kandimalla @anirudh-0

**Comment by subhashini-egov** [comment]
@talele08 - I've suggested enriching the payload in code and pushing to Kafka topic instead of Indexer doing the enrichment. Is there any other solution/workaround to this? If workflow service does sync persisting, will that help this?
